AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/keyrings.smithy

```diff
@@ -340,7 +340,10 @@ structure CreateAwsKmsHierarchicalKeyringInput {
     ttlSeconds: PositiveLong,
 
     @javadoc("Which type of local cache to use.")
-    cache : CacheType
+    cache : CacheType,
+
+    @javadoc("Shared cache.")
+    sharedCache : CacheType
```

Comment:
1. use the documentation trait, not javadoc
2. in documentation, detail that either `cache` or `sharedCache` SHOULD be set, not both
3. in documenation, detail that if both are set, an exception is thrown

Reply:
Thanks